### PR TITLE
Refactoring App Module and App Routing Module

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
This PR includes the implementation of one of the part of ideas proposed in [here](https://github.com/Cloud-CV/GSoC-Ideas/issues/28).

Features added:
1. [app-module](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/app.module.ts) splitted into smaller sub-module
2. [app-routing-module](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/app-routing.module.ts) splitted into smaller sub-routes.

Refactoring Done for following Modules:

-  [x] Auth Module
-  [x] Challenge Module 
-  [x] Dashboard Module
-  [x] Home Module
-  [x] Nav Module
-  [x] Publiclist Module
-  [x] Utility Module

Refactoring TODO list:

-  [ ] Analysis Module
-  [ ] About Module
-  [ ] challenge-create Module
-  [ ] Contact Module
-  [ ] Get-involved Module
-  [ ] Not-found Module
-  [ ] Our-team Module
-  [ ] Privacy-Policy Module
-  [ ] Profile Module

GIF showing that all routes are working fine are refactoring:
ezgif com-video-to-gif (1)

Link to live demo: http://pr-299-evalai.surge.sh

### CODE REVIEW

**Comment by Kajol-Kumari** [comment]
The custom module is working fine for AuthModule but i am unable to get the header and footer in when i used Publiclist custom module. Though i have included "CUSTOM_ELEMENTS_SCHEMA" in the @NgModule.schema but still i am unable to see the header and footer part of the webiste. Any suggestions regarding what am i missing?

cc: @Sanji515 @Shekharrajak  @lunayach @RishabhJain2018

File: src/app/components/home/home.module.ts
Comment (Shekharrajak) [inline src/app/components/home/home.module.ts:26]: HomeComponent uses the headers and footer component, which is not declared in this module. May be we can have `NavModule`  (it will declare the header and footer that will be used across other module as well) which will be imported here.

WDYT ?

File: src/app/components/home/home.module.ts
Comment (Shekharrajak) [inline src/app/components/home/home.module.ts:44]: What does it do ?

**Comment by Shekharrajak** [comment]
Please try to make better PR title and commit message.

File: src/app/components/home/home.module.ts
Comment (Kajol-Kumari) [reply src/app/components/home/home.module.ts:44]: Basically, it defines a schema that will allow any non-Angular elements with 'a -' in their name. Without this, all the selector tags were throwing error of being undefined.

File: src/app/components/home/home.module.ts
Comment (Kajol-Kumari) [reply src/app/components/home/home.module.ts:26]: I had tried that too, but adding the NavModule is making the side-nav and top-nav appear but their routes are not working.

Comment (Kajol-Kumari) [reply src/app/components/home/home.module.ts:26]: @Shekharrajak  I was missing to add the "RouterModule" in the custom modules due to which routerLinks were not getting activated. It's working fine now. I have comitted the required changes. Please look at it once and let me know if any modification is required.

Comment (Shekharrajak) [reply src/app/components/home/home.module.ts:26]: Can you please share the demo link, that surge create during travis CI/CD build ? You can also create some GIF of the working for future reference along with PR description.

**Review by Shekharrajak — COMMENTED** [review]
Thanks for the work! Can you please create a list of modules that you created in this PR and TODO modules that is not cover in this PR? You can create a checklist in PR description or somewhere in different newly opened issue.

**Review by Shekharrajak — APPROVED** [review]
Thanks for GIF and PR description update. May be you can update it in idea list task for future reference. @Sanji515 can you please check it locally once and approve , if it is looking fine ?

**Comment by RishabhJain2018** [comment]
@Sanji515 Can you please take a look at this?
@Kajol-Kumari Can you please create an issue for the remaining work and you can add it in a new PR.

**Review by Sanji515 — APPROVED** [review]
Good Work @Kajol-Kumari 
LGTM :+1: This can be merged :tada:

**Review by lunayach — APPROVED** [review]
Looks good to me, LGTM :tada:

**Comment by Sanji515** [comment]
@Kajol-Kumari I think we can close this PR as the work done in this PR is already included in #303

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari I think we can close this PR as the work done in this PR is already included in #303

yes @Sanji515 agree with you.will close this one.

**Pull request closed without merging**
